crates/chia-consensus/src/merkle_tree.rs

```diff
@@ -123,56 +123,60 @@ fn deserialize_recurse(
 }
 
 impl MerkleSet {
+    pub fn new(leafs: &mut [[u8; 32]]) -> MerkleSet {
```

Comment:
I don't think it makes sense to just add another (unused) function, and more tests for this function. If you would turn `generate_merkle_tree()` into this function, you'd just be renaming it. No need for any new tests